src/VisualStudio/Core/Def/Implementation/ProjectSystem/AbstractProject_IIntellisenseBuildTarget.cs

```diff
@@ -0,0 +1,129 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using Microsoft.Build.Execution;
+using Microsoft.Build.Logging;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Internal.Log;
+using Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem.Interop;
+using Roslyn.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem
+{
+    internal partial class AbstractProject : IIntellisenseBuildTarget
+    {
+        private static readonly object s_diagnosticKey = new object();
+        private static readonly FieldInfo s_fieldInfo = typeof(BuildManager).GetField("_buildParameters", BindingFlags.GetField | BindingFlags.Instance | BindingFlags.NonPublic);
```

Comment:
we can ask them to add API for it, but still not sure why you don't like it. you don't like it because it doesn't benefit users or you don't like it regardless it benefit users or not, it uses non-public API?